### PR TITLE
Bluetooth: host: Log security keys needed by sniffer

### PR DESCRIPTION
**Pull request by joerchan**
Log the security keys that the sniffer needs in order to sucessfully
decrypt the connection.
This option allows the sniffer to work in the cases where enabling
using the SMP debug keys is not wanted, either because it changes the
way the peer behaves or is denied by the peer.
It also enables the sniffer to decrypt a connection where the bond
already exists.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

### CODE REVIEW

**Comment by jhedberg** [comment]
@joerchan looks generally fine to me - why a Draft?

**Review by carlescufi — CHANGES_REQUESTED** [review]
I'm not opposed to this, but you need to update:
https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/kconfig/hardened.csv

**Comment by joerchan** [comment]
> @joerchan looks generally fine to me - why a Draft?

Just a draft while I check that they all work (with the nRF Sniffer) and fix any issues in CI before bothering anyone else :)

The Legacy LTK use-case is an interesting one, because there we have 2 LTKs, and we have to select the one that is distributed by the peripheral. Meaning if they switch role, they also switch LTK. Maybe we shouldn't distribute encryption info in the central role?

@carlescufi The hardened.csv file is be updated for the new option.

**Comment by joerchan** [comment]
@jhedberg @carlescufi I'm thinking about adding the RPA address and renaming to LOG_SNIFFER_INFO since the address is usually how you find which device / connection you should look at. Thoughts?

**Comment by jhedberg** [comment]
> I'm thinking about adding the RPA address and renaming to LOG_SNIFFER_INFO since the address is usually how you find which device / connection you should look at. Thoughts?

@joerchan sounds good to me!

**Comment by carlescufi** [comment]
> @jhedberg @carlescufi I'm thinking about adding the RPA address and renaming to LOG_SNIFFER_INFO since the address is usually how you find which device / connection you should look at. Thoughts?

Great idea, +1

**Comment by joerchan** [comment]
Great, thanks. I have added printing of both RPA and NRPA

File: subsys/bluetooth/host/Kconfig [hunk 1]
```diff
@@ -753,6 +753,17 @@ config BT_DEBUG_SERVICE
 
 endif # BT_DEBUG
 
+config BT_LOG_SNIFFER_INFO
+	bool "Bluetooth log information for sniffer"
```
Comment (jhedberg) [inline subsys/bluetooth/host/Kconfig:757]: Wouldn't it make sense to add `depends on BT_TESTING` here? `BT_TESTING` is intended to make clear which options should not end up in final product configurations.

Comment (joerchan) [reply subsys/bluetooth/host/Kconfig:757]: I'm unsure. BT_TESTING is mostly used for options that changes the behavior, while this only enables logging of certain information.

Looking over the options that we warn on, but don't have depends on BT_TESTING we have the ones that just logs information, but logs security info:
CONFIG_BT_DEBUG_SMP, CONFIG_BT_DEBUG_KEYS, BT_LOG_SNIFFER_INFO.
And the ones that changes behavior:
CONFIG_BT_FIXED_PASSKEY, CONFIG_BT_USE_DEBUG_KEYS, CONFIG_BT_STORE_DEBUG_KEYS

**Review by carlescufi — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**